jerry-core/ecma/operations/ecma-regexp-object.cpp

```diff
@@ -286,7 +286,39 @@ re_match_regexp (re_matcher_ctx_t *re_ctx_p, /**< RegExp matcher context */
         ecma_char_t ch2 = lit_utf8_iterator_read_next (&iter);
         JERRY_DDLOG ("Character matching %d to %d: ", ch1, ch2);
 
-        if (ch1 != ch2)
+        if (re_ctx_p->flags & RE_FLAG_IGNORE_CASE)
+        {
+          ecma_char_t ch1_buffer[LIT_MAXIMUM_OTHER_CASE_LENGTH];
+          ecma_char_t ch2_buffer[LIT_MAXIMUM_OTHER_CASE_LENGTH];
+          lit_utf8_size_t ch1_length = lit_char_to_lower_case (ch1,
+                                                               ch1_buffer,
+                                                               LIT_MAXIMUM_OTHER_CASE_LENGTH);
+
+          lit_utf8_size_t ch2_length = lit_char_to_lower_case (ch2,
+                                                               ch2_buffer,
+                                                               LIT_MAXIMUM_OTHER_CASE_LENGTH);
```

Comment:
The standard explicitly says (in 15.10.2.8 Atom) that you should use the uppercase function. Also, if the length is > 1, the original character should be used (also there is a rule for characters > 127). I think during the pattern construction you should uppercase the pattern characters, so only one call for uppercase could be done here.

Personally I would introduce a regexp_uppercase or some similar helper function. It expects an ecma_char_t and returns with an ecma_char_t. That is all.